File: TextStyling/app/src/androidTest/java/com/android/example/text/styling/renderer/spans/CodeBlockSpanTest.java
package com.android.example.text.styling.renderer.spans;

import static org.junit.Assert.assertEquals;
import static org.mockito.Mockito.mock;

import android.graphics.Color;
import android.graphics.Typeface;
import android.text.TextPaint;

import org.junit.Test;

/**
 * Tests for {@link CodeBlockSpan} class
 */
public class CodeBlockSpanTest {

    private TextPaint paint = mock(TextPaint.class);
    private CodeBlockSpan span = new CodeBlockSpan(Typeface.DEFAULT, Color.RED);

    @Test
    public void updateDrawState() {
        // When the update draw state is called
        span.updateDrawState(paint);

        // Check that the correct color is set
        assertEquals(Color.RED, paint.bgColor);
    }
}

File: TextStylingKotlin/app/src/androidTest/java/com/android/example/text/styling/renderer/spans/CodeBlockSpanTest.kt
package com.android.example.text.styling.renderer.spans

import org.junit.Assert.assertEquals
import org.mockito.Mockito.mock

import android.graphics.Color
import android.graphics.Typeface
import android.text.TextPaint

import org.junit.Test

/**
 * Tests for the [CodeBlockSpan] class
 */
class CodeBlockSpanTest {

    private val paint = mock(TextPaint::class.java)
    private val span = CodeBlockSpan(Typeface.DEFAULT, Color.RED)

    @Test fun updateDrawState() {
        // When the update draw state is called
        span.updateDrawState(paint)

        // Check that the correct color is set
        assertEquals(Color.RED, paint.bgColor)
    }
}